### DISCUSSION TITLE
Support conventional commits

### DISCUSSION THREAD

**Discussion by keikoro** [Zulko/moviepy#2340] [General]
I'd like to propose we introduce [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to increase the specificity of commits/for better differentiation of commits/overall quality of commit messages.

Having had a look at recently submitted PRs, I feel like requiring a commit message `type` would reduce the number of PRs which try to do too much at once because it a) forces people to consciously think about what a commit is trying to achieve and b) makes it immediately obvious when there's a potential lack of separation of concerns (e.g. lots of different types mixed together).

**Comment by keikoro** [comment]
[Gitlint](https://github.com/jorisroovers/gitlint) could be used in a GitHub workflow to enforce CC types, see their [Contrib Rules docs](https://jorisroovers.com/gitlint/latest/rules/contrib_rules/#ct1-contrib-title-conventional-commits).

**Comment by keikoro** [comment]
Conventional commits can also help with better structured release notes, e.g. when used with actions like [release-please](https://github.com/marketplace/actions/release-please-action).

**Comment by OsaAjani** [comment]
I do like the general idea of more standardized commit but I fear this would be yet another obstacle on the path for debutant people just trying to help.
It's always a hard choice between more rules for better uniformity and more freedom for less hassle ^^

**Comment by keikoro** [comment]
Since most PRs we are getting today are at least partially (if not mostly) written by AI anyway, I'm not sure the argument still holds.

At this point, I really think drastic measures are needed to make submissions a lot more focussed and thus reviews better manageable. PRs which combine commits willy-nilly are time-consuming to make sense of either way, but the added vibe code makes it all much worse.

AI shouldn't have a problem with creating 3 small PRs that are to the point instead of one messy large one if those are the rules. And I don't think users who use AI will even think to question AI doing so.
